h/util/db.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import sqlalchemy
+
+
+def on_transaction_end(session):
+    """
+    Decorator for a function which should run after a top-level transaction ended.
+
+    Transaction that are either implicitly or explicity committed or rolled back will be
+    closed at the end of a Pyramid view. This is here for cleaning up caches so that
+    code after the view, exception views for example, will not be able to access
+    detached instances.
+
+    The wrapped function will receive the database session as ``session`` and the
+    transaction as ``transaction``.
+
+    Example usage:
+
+    .. code-block:: python
+
+       @util.db.on_transaction_end(session)
+       def flush_cache(session, transaction):
+           self._cache = {}
+
+    """
+    def wrapped(func):
```

Comment:
The naming here is a bit confusing, because `wrapped` is actually a decorator, not a wrapped function. Calling this `decorate` would probably be a bit clearer, and is [sort of conventional](https://github.com/zzzeek/sqlalchemy/blob/c04870ba7b8098c7d408ad66f60efe7229496fde/lib/sqlalchemy/event/api.py#L92-L126).